patroni/postgresql.py

```diff
@@ -933,6 +976,12 @@ def stop(self, mode='fast', block_callbacks=False, checkpoint=True, on_safepoint
 
     def _do_stop(self, mode, block_callbacks, checkpoint, on_safepoint):
         if not self.is_running():
+            if self.is_running_with_lost_data_dir():
```

Comment:
I think we can get rid from `is_running_with_lost_data_dir`. It is not doing anything useful. The same check is done below.